### PR TITLE
chore: update Storybook components [Design System / Atoms and Molecules]

### PR DESCRIPTION
**Pull request by jessicamcinchak**
Filling in a lot of gaps here, but haven't really sweated what's best categorised as a "Molecule", an "Atom" or otherwise yet - I figure we can revisit that more comprehensively later once we have better component coverage.

My general approach here is:
- Update type annotations & naming in existing stories
- Add new stories for un-documented components
- "Form elements" typically just get a "Basic" story, maybe an additional "With Error" story if applicable

### CODE REVIEW

File: editor.planx.uk/src/components/AnalyticsDisabledBanner.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const meta = {
+  title: "Design System/Molecules/AnalyticsDisabledBanner",
+  component: AnalyticsDisabledBanner,
+} satisfies Meta<typeof AnalyticsDisabledBanner>;
+
+type Story = StoryObj<typeof meta>;
+
+export default meta;
+
+/** Add `&analytics=false` to the end this page's URL to display the banner */
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/components/AnalyticsDisabledBanner.stories.tsx:15]: JSDoc comments translate to Storybook 7 "descriptions" :sparkles: see here for more https://storybook.js.org/docs/react/api/doc-block-description#writing-descriptions

**Comment by jessicamcinchak** [comment]
[Update Storybook](https://trello.com/c/xDSyYBbJ/2290-update-storybook)

File: editor.planx.uk/src/components/AnalyticsDisabledBanner.stories.tsx (hunk 1 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/components/AnalyticsDisabledBanner.stories.tsx:15]: This is a nice feature!

File: editor.planx.uk/src/components/TestEnvironmentBanner.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const meta = {
+  title: "Design System/Molecules/TestEnvironmentBanner",
+  component: TestEnvironmentBanner,
+} satisfies Meta<typeof TestEnvironmentBanner>;
+
+type Story = StoryObj<typeof meta>;
+
+export default meta;
+
+/** Will not display if the page URL includes .uk */
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/components/TestEnvironmentBanner.stories.tsx:15]: 👍

File: editor.planx.uk/src/ui/CollapsibleInput.stories.tsx [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
   component: CollapsibleInput,
-};
+} satisfies Meta<typeof CollapsibleInput>;
 
-export const Basic: StoryObj<Props> = {
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+// TODO clarify use of args.children versus render here
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/ui/CollapsibleInput.stories.tsx:16]: Yeah I also had this question whilst reviewing - it looks like `render()` is actually optional and will be defaulted using these `args` and the predefined `meta` right?

**Review by DafyddLlyr — APPROVED** [review]
Good to get these converted / updated, and lots of new additions also.

I've not necessarily been thinking in terms of atoms / molecules often when building components (often as many of the building blocks have already been created), but it's quite a helpful framework to have. I agree with the approach of revisiting (and defining) these categories later once we have Storybook updated as we want it.

**Pull request merged**